desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -116,7 +116,8 @@ void DownloadFromDCWidget::updateProgressBar()
 		ui.progressBar->setFormat("%p%");
 	}
 	ui.progressBar->setValue(lrint(progress_bar_fraction * 100));
-	free(last_text);
+	if (last_text)
+		free(last_text);
 	last_text = strdup(progress_bar_text);
```

Comment:
the crash in #507 is in strdup... is it possible that calling strdup on NULL is what causes the crash?